### PR TITLE
Make graphviz a soft dependency

### PR DESCRIPTION
**Pull request by gregorybchris**
Makes graphviz a soft dependency and part of the `plt` extra. Export to graphviz is not essential causal functionality.

### CODE REVIEW

**Comment by gregorybchris** [comment]
Note: there are also short-term benefits to removing graphviz as a hard dependency due to a bug in conda resulting in environment creating failures when pip's graphviz and the conda package python-graphviz conflict: https://github.com/conda/conda/issues/10742

**Review by gaugup — APPROVED** [review]

**Review by kbattocchi — CHANGES_REQUESTED** [review]
Mostly looks good, except for one minor issue listed below.

File: econml/_tree_exporter.py [hunk 1]
```diff
@@ -28,6 +27,15 @@
...
 
+try:
+    import graphviz
+except ImportError as exn:
+    from .utilities import MissingModule
+
+    # make any access to graphviz or plt throw an exception
+    graphviz = plt = MissingModule("graphviz is no longer a dependency of the main econml package; "
+                                   "install econml[plt] or econml[all] to require it, or install graphviz "
+                                   "separately, to use the tree interpreters", exn)
```
[the remark this answers is not in the document]
Comment (gregorybchris) [reply econml/_tree_exporter.py]: Oh yikes.. thanks

**Review by kbattocchi — APPROVED** [review]

**Comment by kbattocchi** [comment]
@gregorybchris The notebook test failure is real - you'll now need to add `plt` to the package extras (in addition to `tf`) here: https://github.com/microsoft/EconML/blob/d5639db2ebd1a04e6fe8d4d44c15c0f976305b08/azure-pipelines.yml#L89

File: azure-pipelines.yml [hunk 2]
```diff
@@ -86,7 +86,7 @@ jobs:
   parameters:
     versions: ['3.8']
     images: ['ubuntu-18.04']
-    package: '-e .[tf]'
+    package: '-e .[tf, plt]'
```
Comment (kbattocchi) [inline azure-pipelines.yml:89]: The space is causing pip to treat `plt]` as a separate requirement.

```suggestion
    package: '-e .[tf,plt]'
```

**Pull request merged**